src/Umbraco.Core/Services/MemberService.cs

```diff
@@ -761,7 +766,11 @@ public void Save(IMember member)
 
             _memberRepository.Save(member);
 
-            scope.Notifications.Publish(new MemberSavedNotification(member, evtMsgs).WithStateFrom(savingNotification));
+            if (publishNotificationSaveOptions is PublishNotificationSaveOptions.All or PublishNotificationSaveOptions.Saved)
+            {
+                var savingNotification = new MemberSavingNotification(member, evtMsgs);
```

Comment:
I believe rather than creating a new one here, you should be using the one you have created earlier (so the state can be passed on).  Which it tricky now you may or may not have created and published it of course.

But you need to track if you've created a `Saving` notification, and then when you create the `Saved` one only call the `.WithStateFrom(...)` if you have a `Saving` notification created already,

Reply:
Done 🚀